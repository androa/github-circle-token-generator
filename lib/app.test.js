const request = require("supertest");
const fs = require("fs");
const { get } = require("axios");
const { setGitHubTokenOnProject } = require("./circleci");

jest.mock("axios");
jest.mock("./circleci");

process.env.GITHUB_APP_ID = "123";
process.env.GITHUB_APP_KEY = fs.readFileSync(
  "./lib/__mocks__/github-app-key.pem"
);

const app = require("./app");

describe("GET /token/inject", () => {
  it("adds token to environment of CircleCI project", done => {
    setGitHubTokenOnProject.mockResolvedValue({ name: "BAR", value: "xxxFOO" });

    request(app)
      .get("/token/inject?username=test&project=test")
      .expect(({ text }) => {
        expect(text).toMatchSnapshot();
      })
      .expect(200, done);
  });

  it("fails if username is missing", done => {
    request(app)
      .get("/token/inject?project=test")
      .expect(({ body }) => {
        expect(body).toMatchSnapshot();
      })
      .expect(400, done);
  });

  it("fails if project is missing", done => {
    request(app)
      .get("/token/inject?username=test")
      .expect(({ body }) => {
        expect(body).toMatchSnapshot();
      })
      .expect(400, done);
  });

  it("fails if username and project is missing", done => {
    request(app)
      .get("/token/inject")
      .expect(({ body }) => {
        expect(body).toMatchSnapshot();
      })
      .expect(400, done);
  });
});

describe("GET /token/verify", () => {
  it("tests the token generation against the GitHub API", done => {
    get.mockResolvedValue({ data: { id: "123" } });

    request(app)
      .get("/token/verify")
      .expect(({ text }) => {
        //expect(text).toMatchSnapshot();
      })
      .expect(200, done);
  });
});
