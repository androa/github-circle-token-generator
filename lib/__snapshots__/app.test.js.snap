// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /token/inject adds token to environment of CircleCI project 1`] = `"Successfully created token (ending with xxxFOO) and injected it to CircleCI build (test/test)"`;

exports[`GET /token/inject fails if project is missing 1`] = `
Object {
  "errors": Array [
    Object {
      "location": "query",
      "msg": "Project identifier is required",
      "param": "project",
    },
  ],
}
`;

exports[`GET /token/inject fails if username and project is missing 1`] = `
Object {
  "errors": Array [
    Object {
      "location": "query",
      "msg": "Username is required",
      "param": "username",
    },
    Object {
      "location": "query",
      "msg": "Project identifier is required",
      "param": "project",
    },
  ],
}
`;

exports[`GET /token/inject fails if username is missing 1`] = `
Object {
  "errors": Array [
    Object {
      "location": "query",
      "msg": "Username is required",
      "param": "username",
    },
  ],
}
`;

exports[`GET /token/verify tests the token generation against the GitHub API 1`] = `"{\\"response\\":\\"Got reply from github\\"}"`;
